src/lib/search-url-builder.ts

```diff
@@ -31,11 +31,12 @@ const toTitleCase = (name: string) => {
 }
 
 const tagsForPath = (path: string) => {
-  const tagsSplit = path?.split('-lessons-by-') || []
+  const [tagsString] = path?.split('-lessons-by-') ?? []
 
-  const tags = tagsSplit.length >= 1 ? tagsSplit[0].split('-and-').sort() : []
+  if (isUndefined(tagsString)) return []
+  if (tagsString.startsWith('lessons-by')) return []
```

Comment:
This `startsWith` fixes the main issue. 

The other changes make sure we return an empty Array instead of `undefined` so the return type is always an Array of Strings.